msm/msm

```diff
@@ -23,114 +23,166 @@
+            echo "Unable to pull master skills list!"
+            exit 111    
+    fi
+else 
+    if ! wget -qO- "https://raw.githubusercontent.com/MycroftAI/mycroft-skills/master/.gitmodules" ; then
+            echo "Unable to pull master skills list!"
+            exit 112
+    fi
+fi
 }
 
 function install() {
-        goto_skills_dir
-	if [ -z "$2" ]; then
-		echo "You must pass the git url or skill name"
-		exit 1
-	fi
-	if [[ "$2" == "git@"* || "$2" == "https://"* || "$2" == "http://"* ]]; then
-		repo=$2
-	else
-		skill_list="`curl -s "https://raw.githubusercontent.com/MycroftAI/mycroft-skills/master/.gitmodules"`"
-		skills=`echo "$skill_list" | grep -n 'submodule' | sed 's/[[:space:]]//g' | sed 's/\[submodule"//g' | sed 's/"\]//g'`
-		exact_match=`echo "$skills" | grep -i ".*:$2$"`
-		skill=`echo "$skills" | grep -i ".*:.*$2.*"`
-		if [ ! -z $exact_match ]; then
-			skill=$exact_match
-		fi
-		git_line=`echo "$skill" | sed 's/\:.*//'`
-
-		if [[ $skill ==  *$'\n'* ]]; then
-			echo -e "Your search has multiple choices\n\n$skill" | sed 's/.*://g'
-			exit 2
-		else
-			if [ -z $git_line ]; then
-				echo "Skill not found"
-				exit 3
-			fi
-			repo_line=$(($git_line + 2))
-			repo=`echo "$skill_list" | sed -n $repo_line'{p;q;}' | sed 's/[[:space:]]//g' | sed 's/url=//g'`
-		fi 
-	fi
-	git_name=`echo "$repo" | sed 's/.*\///'`
-	name=`echo "$git_name" | sed 's/.git//'`
-        echo "Cloning repository"
-        git clone $repo >> /dev/null
-        cd $name
-         if [ -f "requirements.txt" ]; then
-         	echo "Installing libraries requirements"
-        	pip install -r requirements.txt
+cd "${mycroft_skill_folder}"
+if [[ ${no_vwrap} == 0 ]] ; then
+    echo "Missing virtualwrapper, cowardly refusing to install skills."
+    return 5
+fi
+# loop through list of arguments
+while [[ $# -gt 0 ]] ; do
+    iskill="${1}";
+    shift;
+    echo "Attempting to install ${iskill}..."
+    if [[ "${iskill}" == "git@"* || "${iskill}" == "https://"* || "${iskill}" == "http://"* ]]; then
+        repo="${iskill}"
+    else
+        skills=$(list | grep -n 'submodule' | sed 's/[[:space:]]//g' | sed 's/\[submodule"//g' | sed 's/"\]//g')
+        exact_match=$(echo "$skills" | grep -i ".*:${iskill}$")
+        skill=$(echo "$skills" | grep -i ".*:.*${iskill}.*")
+        if [[ ! -z "${exact_match}" ]]; then
+            skill=${exact_match}
+        fi
+        git_line=$(echo "$skill" | sed 's/\:.*//')
+    
+        if [[ "${skill}" ==  *$'\n'* ]]; then
+            echo -e "Your search has multiple choices\n\n$skill" | sed 's/.*://g'
+            return 3
+        else
+            if [[ -z "${git_line}" ]]; then
+                echo "A ${iskill} skill was not found"
+                return 3
+            fi
+        repo_line=$(($git_line + 2))
+        repo=$(echo "$skill_list" | sed -n $repo_line'{p;q;}' | sed 's/[[:space:]]//g' | sed 's/url=//g')
```

Comment:
`skill_list` is missing, I think this is what makes the installation of named skills fail.